interpreters/lua/Make.defs

```diff
@@ -0,0 +1,31 @@
+############################################################################
+# apps/interpreters/lua/Make.defs
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.  The
+# ASF licenses this file to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance with the
+# License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+############################################################################
+
+ifneq ($(CONFIG_INTERPRETERS_LUA),)
+CONFIGURED_APPS += $(APPDIR)/interpreters/lua
+
+# Enable <lua.h> and <lauxlib.h> includes.
+
+CFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" \
+	$(APPDIR)/interpreters/lua/lua-$(CONFIG_INTERPRETER_LUA_VERSION)/src}
+CXXFLAGS += ${shell $(INCDIR) $(INCDIROPT) "$(CC)" \
```

Comment:
it's very easy to introduce the conflict if we expose the whole code base to the global search path.

Reply:
I'm following the Make.defs pattern of other interpreters (quickjs, wasm3, wamr, duktape) and libraries (libuv, mbedtls, cmsis). Is there a recommendation for an alternative approach?  All Lua headers are prefixed with an `l` character, so they should hopefully not cause any conflicts.

Reply:
If so, it's fine to add into the search path.